### PR TITLE
Support SendCroToIbc in evm_log_handlers

### PR DESCRIPTION
**Pull request by thomas-nguy**
Linked to issue #68 

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: x/cronos/keeper/evm_log_handlers_test.go [hunk 1]
```diff
@@ -297,3 +297,82 @@ func (suite *KeeperTestSuite) TestSendToIbcHandler() {
...
+				suite.Require().NoError(err)
+				data = input
+			},
+			func() {},
+			errors.New("0aphoton is smaller than 10000000000000aphoton: insufficient funds"),
+		},
+		{
+			"success send cro to ibc",
+			func() {
+				suite.app.CronosKeeper.SetExternalContractForDenom(suite.ctx, suite.evmParam.EvmDenom, contract)
```
Comment (yihuang) [inline x/cronos/keeper/evm_log_handlers_test.go:328]: I think we don't need the mapping for this to work?

File: x/cronos/keeper/evm_log_handlers_test.go [hunk 2]
```diff
@@ -297,3 +297,82 @@ func (suite *KeeperTestSuite) TestSendToIbcHandler() {
...
+				err := suite.MintCoins(sdk.AccAddress(contract.Bytes()), sdk.NewCoins(coin))
+				suite.Require().NoError(err)
+
+				balance := suite.app.BankKeeper.GetBalance(suite.ctx, sdk.AccAddress(contract.Bytes()), suite.evmParam.EvmDenom)
+				suite.Require().Equal(coin, balance)
+
+				// Mint coin for the module
+				suite.MintCoinsToModule(types.ModuleName, sdk.NewCoins(sdk.NewCoin(types.IbcCroDenomDefaultValue, sdk.NewInt(123))))
+
+				err = suite.MintCoins(sdk.AccAddress(contract.Bytes()), sdk.NewCoins(coin))
```
Comment (yihuang) [inline x/cronos/keeper/evm_log_handlers_test.go:339]: Why mint twice?

File: x/cronos/keeper/evm_log_handlers_test.go [hunk 3]
```diff
@@ -297,3 +297,82 @@ func (suite *KeeperTestSuite) TestSendToIbcHandler() {
...
+				suite.Require().NoError(err)
+
+				input, err := keeper.SendToIbcEvent.Inputs.Pack(
+					"recipient",
+					coin.Amount.BigInt(),
+				)
+				data = input
+			},
+			func() {},
```
Comment (yihuang) [inline x/cronos/keeper/evm_log_handlers_test.go:348]: I think it is worth doing the post checking that:
- balance of the contract is good
- balance of the module account is good

**Review by yihuang — APPROVED** [review]
LGTM

**Comment by thomas-nguy** [comment]
Tested locally and it works 👌

**Pull request merged**